explorer/Explorer.tsx

```diff
@@ -351,20 +357,68 @@ export class Explorer
         } = grapherConfigFromExplorer
         const grapherConfigFromExplorerOnlyGrapherProps = omit(
             grapherConfigFromExplorer,
-            ["mapTargetTime"]
+            [
+                "yIndicatorIds",
+                "xIndicatorId",
+                "colorIndicatorId",
+                "sizeIndicatorId",
+                "mapTargetTime",
+            ]
         )
 
-        const hasGrapherId = grapherId && isNotErrorValue(grapherId)
+        // chart creation preference: with grapher id -> with indicator ids -> with manually provided data
+        let creationMode = ChartCreationMode.Unknown
+        if (grapherId && isNotErrorValue(grapherId))
+            creationMode = ChartCreationMode.WithGrapherId
+        else if (yIndicatorIds?.length)
+            creationMode = ChartCreationMode.WithIndicatorIds
+        else if (tableSlug)
+            creationMode = ChartCreationMode.WithManuallyProvidedData
 
-        const grapherConfig = hasGrapherId
-            ? this.grapherConfigs.get(grapherId!) ?? {}
-            : {}
+        const grapherConfig =
+            creationMode === ChartCreationMode.WithGrapherId
+                ? this.grapherConfigs.get(grapherId!) ?? {}
+                : {}
 
         const config: GrapherProgrammaticInterface = {
             ...grapherConfig,
             ...grapherConfigFromExplorerOnlyGrapherProps,
             hideEntityControls: this.showExplorerControls,
-            manuallyProvideData: tableSlug ? true : false,
+            manuallyProvideData:
+                creationMode === ChartCreationMode.WithManuallyProvidedData,
+        }
+
+        // set given indicators as dimensions
+        if (creationMode === ChartCreationMode.WithIndicatorIds) {
+            const dimensions = config.dimensions ?? []
+            if (yIndicatorIds) {
+                yIndicatorIds.forEach((yIndicatorId) => {
+                    dimensions.push({
+                        variableId: yIndicatorId,
+                        property: DimensionProperty.y,
+                    })
+                })
+            }
+            if (xIndicatorId) {
+                dimensions.push({
+                    variableId: xIndicatorId,
+                    property: DimensionProperty.x,
+                })
+            }
+            if (colorIndicatorId) {
+                dimensions.push({
+                    variableId: colorIndicatorId,
+                    property: DimensionProperty.color,
+                })
+            }
+            if (sizeIndicatorId) {
+                dimensions.push({
+                    variableId: sizeIndicatorId,
+                    property: DimensionProperty.size,
+                })
+            }
+            config.dimensions = dimensions
+            config.bakedGrapherURL = "/grapher"
```

Comment:
Should I use an existing constant here, maybe from the [client settings](https://github.com/owid/owid-grapher/blob/1007f016506b4dfeae7fa71064b0682316a997f0/settings/clientSettings.ts)?